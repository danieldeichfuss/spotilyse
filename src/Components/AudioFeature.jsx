import React from "react";
import Bar from "./Bar";
import Guitar from "../img/Guitar";
import Robot from "../img/Robot";
import Dancer from "../img/Dancer";
import Stone from "../img/Stone";
import Thunder from "../img/Thunder";
import Sunglasses from "../img/Sunglasses";
import Microphone from "../img/Microphone";
import Headphones from "../img/Headphones";
import Cloud from "../img/Cloud";
import Ear from "../img/Ear";
import Bunny from "../img/Bunny";
import Snail from "../img/Snail";
import Sad from "../img/Sad";
import Sun from "../img/Sun";

const AudioFeature = ({ audioFeature, featureValue }) => {
  // NOTE: change < to > to see all the other
  const message =
    featureValue < audioFeature.breakPoint
      ? audioFeature.messages[1]
      : audioFeature.messages[0];
  let img = null;
  img = <Robot />;

  // TODO: Find a better solution
  switch (message.img) {
    case "guitar":
      img = <Guitar />;
      break;
    case "robot":
      img = <Robot />;
      break;
    case "dancer":
      img = <Dancer />;
      break;
    case "stone":
      img = <Stone />;
      break;
    case "thunder":
      img = <Thunder />;
      break;
    case "sunglasses":
      img = <Sunglasses />;
      break;
    case "microphone":
      img = <Microphone />;
      break;
    case "headphones":
      img = <Headphones />;
      break;
    case "ear":
      img = <Ear />;
      break;
    case "cloud":
      img = <Cloud />;
      break;
    case "bunny":
      img = <Bunny />;
      break;
    case "snail":
      img = <Snail />;
      break;
    case "sun":
      img = <Sun />;
      break;
    case "sad":
      img = <Sad />;
      break;
    default:
      img = null;
  }

  let featureValueProcessed = featureValue;

  if (audioFeature.unit === "%") {
    featureValueProcessed = featureValue * 100;
  }

  featureValueProcessed = Math.floor(featureValueProcessed) + audioFeature.unit;

  return (
    <div className="AudioFeature">
      <div className="AudioFeature__intro-wrapper">
        <h3 className="AudioFeature__title">{audioFeature.name}:</h3>
        <span className="AudioFeature__value">{featureValueProcessed}</span>
      </div>
      {audioFeature.unit === "%" && <Bar value={featureValueProcessed} />}
      <div className="AudioFeature__message">{message.text}</div>
      <div
        className="AudioFeature_img"
        style={{
          margin: "auto",
          width: "150px"
        }}
      >
        {img}
      </div>
    </div>
  );
};

export default AudioFeature;
