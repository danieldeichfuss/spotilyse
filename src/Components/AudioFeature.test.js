import React from "react";
import AudioFeature from "./AudioFeature";
import * as Enzyme from "enzyme";
import ReactSixteenAdapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new ReactSixteenAdapter() });

const audioFeature = {
  messages: ["Message 1", "Message 2"]
};
const featureValue = 0.5;
const wrapper = Enzyme.shallow(
  <AudioFeature audioFeature={audioFeature} featureValue={featureValue} />
);

it("renders page correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
