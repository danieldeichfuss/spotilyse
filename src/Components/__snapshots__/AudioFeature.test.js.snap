// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AudioFeature
    audioFeature={
        Object {
            "messages": Array [
              "Message 1",
              "Message 2",
            ],
          }
    }
    featureValue={0.5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="AudioFeature__intro-wrapper"
>
          <h3
                    className="AudioFeature__title"
          >
                    :
          </h3>
          <span
                    className="AudioFeature__value"
          />
</div>,
        false,
        <div
          className="AudioFeature__message"
/>,
        <div
          className="AudioFeature_img"
          style={
                    Object {
                              "margin": "auto",
                              "width": "150px",
                            }
          }
/>,
      ],
      "className": "AudioFeature",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="AudioFeature__title"
>
              :
</h3>,
            <span
              className="AudioFeature__value"
/>,
          ],
          "className": "AudioFeature__intro-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                ":",
              ],
              "className": "AudioFeature__title",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              ":",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": NaN,
              "className": "AudioFeature__value",
            },
            "ref": null,
            "rendered": NaN,
            "type": "span",
          },
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "AudioFeature__message",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "AudioFeature_img",
          "style": Object {
            "margin": "auto",
            "width": "150px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="AudioFeature__intro-wrapper"
>
            <h3
                        className="AudioFeature__title"
            >
                        :
            </h3>
            <span
                        className="AudioFeature__value"
            />
</div>,
          false,
          <div
            className="AudioFeature__message"
/>,
          <div
            className="AudioFeature_img"
            style={
                        Object {
                                    "margin": "auto",
                                    "width": "150px",
                                  }
            }
/>,
        ],
        "className": "AudioFeature",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="AudioFeature__title"
>
                :
</h3>,
              <span
                className="AudioFeature__value"
/>,
            ],
            "className": "AudioFeature__intro-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  ":",
                ],
                "className": "AudioFeature__title",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                ":",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": NaN,
                "className": "AudioFeature__value",
              },
              "ref": null,
              "rendered": NaN,
              "type": "span",
            },
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "AudioFeature__message",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "AudioFeature_img",
            "style": Object {
              "margin": "auto",
              "width": "150px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
